File: food-review/src/main/java/com/food/review/gateway/mongo/ReviewGatewayImpl.java
package com.food.review.gateway.mongo;

import com.food.review.domain.Review;
import com.food.review.gateway.ReviewGateway;
import com.food.review.gateway.mongo.entity.ReviewMongo;
import com.food.review.gateway.mongo.repository.ReviewRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.UUID;

@Component
public class ReviewGatewayImpl implements ReviewGateway {

    private final ReviewRepository reviewRepository;

    @Autowired
    public ReviewGatewayImpl(ReviewRepository reviewRepository) {
        this.reviewRepository = reviewRepository;
    }

    @Override
    public Review create(Review review) {
        ReviewMongo reviewMongo = ReviewMongo.builder()
                .id(UUID.randomUUID())
                .orderId(review.getOrderId())
                .username(review.getUsername())
                .name(review.getName())
                .text(review.getText())
                .createdAt(review.getCreatedAt())
                .build();

        ReviewMongo savedReviewMongo = this.reviewRepository.save(reviewMongo);

        return review.toBuilder()
                .id(savedReviewMongo.getId())
                .build();
    }
}

File: food-review/src/main/kotlin/com/food/review/gateway/mongo/ReviewGatewayImpl.kt
package com.food.review.gateway.mongo

import com.food.review.domain.Review
import com.food.review.gateway.ReviewGateway
import com.food.review.gateway.mongo.entity.ReviewMongo
import com.food.review.gateway.mongo.mapper.ReviewMongoMapper
import com.food.review.gateway.mongo.repository.ReviewRepository
import org.springframework.stereotype.Component

@Component
class ReviewGatewayImpl(private val reviewRepository: ReviewRepository): ReviewGateway {

    override fun save(review: Review): Review {
        val reviewMongo = ReviewMongoMapper.INSTANCE.map(review)
        val reviewMongoSaved = this.reviewRepository.save(reviewMongo)
        return ReviewMongoMapper.INSTANCE.map(reviewMongoSaved)
    }
}
